lint.go

```diff
@@ -713,3 +714,83 @@ func CheckEarlyDefer(f *lint.File) {
 	}
 	f.Walk(fn)
 }
+
+func CheckEmptyCriticalSection(f *lint.File) {
+	fn := func(node ast.Node) bool {
+		block, ok := node.(*ast.BlockStmt)
+		if !ok {
+			return true
+		}
+		if len(block.List) < 2 {
+			return true
+		}
+		for i := range block.List {
+			if i == len(block.List)-1 {
+				break
+			}
+
+			mutexParams := func(s ast.Stmt) ([]string, string, bool) {
```

Comment:
Use named return parameters. From the current signature, it is not clear at all what this function returns.